README.ideal.md

```diff
@@ -0,0 +1,145 @@
+_Ni-Mh — batteries included environments with Nickel inside_.
+
+Managing a project's development environment involves configuring a lot of tools and occasionnally getting to communicate with each other: One (or several) package manager, a CI system, a service manager (to run that local postgresql database without which you can't test anything), custom utility scripts that end up spawning everywhere, etc.
+Ni-Mh (called “nickel-nix” or “Nixel” at the moment, but I really like Ni-Mh) aims at being your main entrypoint for managaing all these different tools, so that you can get:
+
+1. A unified configuration framework for all of these
+2. A powerful and ergonomic language to allow you to easily abstract over these configurations, with discoverability and early error reporting
+
+## Getting started
+
+To start using `nickel-nix`, you need `Nix` to be installed and configured.
+If it isn't already the case, you can get it with:
+
+```console
+$ curl -L https://nixos.org/nix/install | bash
+# We also need a couple of experimental Nix features
+$ mkdir -p ~/.config/nix
+$ echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf
+```
+
+Then bootstrap a project with:
+
+```console
+$ nix flake init -t github:nickel-lang/nickel-nix
+# Edit the project file to fit your needs
+$ $EDITOR project.ncl
+# Enter the environment
+$ nix develop
+```
+
+## Managing your environment with Nixel
+
+### Dependencies
+
+Nixel can be used to declare the dependencies for your project.
+These can then be instantiated using [Nix](https://nixos.org/nix).
+
+> TODO: Document
+
+### CI configuration
+
+You can use Nixel to generate a CI configuration.
+This is extremely handy if you want to test some non-trivial matrix of platforms/configuration.
+
+Only Github actions is supported at the moment, but we plan to add more options soon.
+
+> TODO: Document
+
+### Development services
+
+Nixel can also be used for some simple service management.
+In combination with the Nix integration, this allows you to quickly spawn any service that you might depend on.
+
+Once the services have been defined in your `project.ncl` file, you can run `nixel up` to start them.
+
+> TODO: Document
+
+### Deployment
+
+If you want to push things yet a bit further, Nixel can also interoperate with [tf-ncl](https://github.com/tweag/tf-ncl) to allow you to manage the deployment infrastructure for your project.
```

Comment:
I think we should say something like "we provide a way to describe environments that can use things like honcho for local dev deployment or tf-ncl for production deployment".